src/general/Species.cs

```diff
@@ -126,6 +126,21 @@ public void ApplyImmediatePopulationChange(long constant, float coefficient)
             Population = 0;
     }
 
+    public void ApplyImmediatePopulationChangeInPatch(long constant, float coefficient, Patch patch)
+    {
+        var oldPopulation = patch.GetSpeciesPopulation(this);
```

Comment:
This method should throw an exception if used for something else than the player. As that should not be allowed as I think that'll make it more likely for auto-evo to fail with a collection modified error.